src/command.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Custom commands.
+
+use std::any::Any;
+use std::sync::Arc;
+
+/// An identifier for a particular command.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Selector(&'static str);
+
+/// An arbitrary command.
+///
+/// A `Command` consists of a `Selector`, that indicates what the command is,
+/// and an optional argument, that can be used to pass arbitrary data.
+///
+/// # Examples
+/// ```
+/// use druid::{Command, Selector};
+///
+/// let selector = Selector::new("process_rows");
+/// let rows = vec![1, 3, 10, 12];
+/// let command = Command::new(selector, rows);
+///
+/// assert_eq!(command.get_object(), Some(&vec![1, 3, 10, 12]));
+/// ```
+#[derive(Debug, Clone)]
+pub struct Command {
+    pub selector: Selector,
+    pub object: Option<Arc<dyn Any>>,
```

Comment:
Discussion question: is the option dynamic or static (ie some selectors just don't need arguments)? In the latter case, it might be a convention to use `()` as the argument. On the other hand, that's maybe less efficient, as it would still need to be boxed...

Reply:
My intention was that the expected type of the argument is determined by the selector. Also I'm not sure if this needs to be `pub`; it should only be accessed through `get_object`, I think..